src/mscorlib/src/System/Reflection/AssemblyName.cs

```diff
@@ -533,7 +571,83 @@ internal string GetNameWithPublicKey()
         private extern byte[] nGetPublicKeyToken();
     
         [System.Security.SecurityCritical]  // auto-generated
-        [MethodImplAttribute(MethodImplOptions.InternalCall)]
-        static internal extern String EscapeCodeBase(String codeBase);
+        static internal String EscapeCodeBase(String codebase)
+        {
+            return EscapeString(codebase, false, true, true);
```

Comment:
It looks like the desktop framework is not escaping #, [, or ]. It's also not escaping the % when what follows makes it not a hex escape. For instance, "%hi" or "%1g"